tenacity/_asyncio.py

```diff
@@ -30,6 +31,15 @@
 WrappedFn = t.TypeVar("WrappedFn", bound=t.Callable[..., t.Awaitable[t.Any]])
 
 
+def is_coroutine_callable(call: t.Callable[..., t.Any]) -> bool:
```

Comment:
Could we have unit tests around this?

Reply:
Good call! I ended up moving the function to `_utils` and slightly amending it (odd `partial` behaviour in 3.8 and 3.9, even though [docs say otherwise](https://docs.python.org/3/library/inspect.html#inspect.iscoroutinefunction)). wdyt? https://github.com/jd/tenacity/pull/437/commits/2616c4ad586f91b9cbad4e777f177fcc199911d2